spec/ext/active_record_spec.rb

```diff
@@ -98,6 +98,27 @@
     end
   end
 
+  describe 'Linked records' do
+    before do
+      @ap = AwesomePrint::Inspector.new(:plain => true)
+    end
+
+    it 'should show the entire record' do
+      e = Email.create(:email_address => 'foo@bar.com')
+      User.last.emails << e
+      email_record = User.joins(:emails).select('emails.email_address').last
+      out = @ap.awesome(email_record)
+      raw_object_string = <<~EOS.strip
+        #<User:placeholder_id> {
+                       "id" => nil,
+            "email_address" => "#{e.email_address}"
+        }
+      EOS
+
```

Comment:
Sorry squiggly heredocs were only added in the latest versions of ruby, so you will need to use this instead:

```ruby
      raw_object_string = <<-EOS.strip
#<User:placeholder_id> {
               "id" => nil,
    "email_address" => "#{e.email_address}"
}
EOS

```

You can copy the format of the other docblocks in the file and then you can test it works by running `bundle exec appraisal rails-4.2 rspec spec/ext/active_record_spec.rb` after setting your ruby version to a version of Ruby earlier than 2.3.0.

e.g. `chruby 1.9.3` or whatever you use to manage your Ruby versions.